docs/source/modules/zos_stat.rst

```diff
@@ -0,0 +1,670 @@
+
+  | **required**: False
+  | **type**: bool
+  | **default**: True
+
+
+tmp_hlq
+  Override the default high level qualifier (HLQ) for temporary data sets.
+
+  The default HLQ is the Ansible user used to execute the module and if that is not available, then the environment variable value ``TMPHLQ`` is used.
+
+  | **required**: False
+  | **type**: str
+
+
+
+
+Examples
+--------
+
+.. code-block:: yaml+jinja
+
+   
+   - name: Get the attributes of a sequential data set on volume '000000'.
+     zos_stat:
+       name: USER.SEQ.DATA
+       type: data_set
+       volume: "000000"
+
+   - name: Get the attributes of a PDSE managed by SMS.
+     zos_stat:
+       name: USER.PDSE.DATA
+       type: data_set
+       volume: "000000"
+       sms_managed: true
+
+   - name: Get the attributes of a sequential data set with a non-default temporary HLQ.
+     zos_stat:
+       name: USER.SEQ.DATA
+       type: data_set
+       volume: "000000"
+       tmp_hlq: "RESTRICT"
+
+   - name: Get the attributes of a generation data group.
+     zos_stat:
+       name: "USER.GDG.DATA"
+       type: data_set
+       volume: "000000"
+
+   - name: Get the attributes of a generation data set.
+     zos_stat:
+       name: "USER.GDG.DATA(-1)"
+       type: data_set
+       volume: "000000"
+
+
+
+
+Notes
+-----
+
+.. note::
+   When querying data sets, the module will create a temporary data set that requires around 4 kilobytes of available space on the remote host. This data set will be removed before the module finishes execution.
+
+   Sometimes, the system could be unable to properly determine the organization or record format of the data set or the space units used to represent its allocation. When this happens, the values for these fields will be null.
+
+   When querying a partitioned data set (PDS), if the Ansible user has RACF READ authority on it, the last referenced date will be updated by the query operation.
+
+
+
+See Also
+--------
+
+.. seealso::
+
+   - :ref:`ansible.builtin.stat_module`
+   - :ref:`zos_find_module`
+
+
+
+
+Return Values
+-------------
+
+
+stat
+  Dictionary containing information about the resource.
+
+  | **returned**: success
+  | **type**: dict
+
+  name
+    Name of the resource queried.
+
+    For Generation Data Sets (GDSs), this will be the absolute name.
+
+    | **returned**: success
+    | **type**: str
+    | **sample**: USER.SEQ.DATA.SET
+
+  resource_type
+    One of 'data_set', 'file' or 'aggregate'.
+
+    | **returned**: success
+    | **type**: str
+    | **sample**: data_set
+
+  attributes
+    Dictionary containing all the stat data.
+
+    | **returned**: success
+    | **type**: dict
+
+    dsorg
+      Data set organization.
+
+      | **returned**: success
+      | **type**: str
+      | **sample**: PS
+
+    type
+      Type of the data set.
+
+      | **returned**: success
+      | **type**: str
+      | **sample**: LIBRARY
+
```

Comment:
I think this should be lower case ("library" instead of "LIBRARY").

Reply:
Thanks for reminding me of this, I changed the code last week so the module would return everything in lowercase, but I forgot about updating the documentation too.